component/file_cache/file_cache.go

```diff
@@ -63,7 +63,7 @@ type FileCache struct {
 
 	createEmptyFile bool
 	allowNonEmpty   bool
-	cacheTimeout    uint32
+	cacheTimeout    float64
```

Comment:
Why are we making this a float?

Reply:
the time comparison part needs float so if we do not modify the base type here, every comparison will need type conversion.